src/main/java/MisterDuke.java

```diff
@@ -0,0 +1,678 @@
+                tempTaskList.add(task);
+            }
+        }
+        ui.showMatchingItems(tempTaskList);
+    }
+
+    public boolean isRunning() {
+        return true;
+    }
+}
+
+/**
+ * The Task class is a parent class for ToDo, Deadline and Event
+ * whereby each Task has a boolean to indicate whether it has been done
+ */
+class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * The Task applies to inputs with sufficient information to classify as a todo, deadline
+     * or event task.
+     * @param description describes the details of a task supplied to the chat bot
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * For a given task, indicate whether it has been done using a space or a tick
+     * @return if the task has been done, return a tick, else return a space
+     */
+    public String getStatusIcon() {
+        return (isDone ? "\u2713" : " "); //return tick symbol
+    }
+
+    /**
+     * For a given task, if it is done, mark it as done by changing the boolean isDone to true
+     */
+    public void markAsDone() {
+        isDone = true;
+    }
+
+    @Override
+    public String toString() {
+        return "[" + getStatusIcon() + "] " + description;
+    }
+}
+
+/**
+ * The ToDo class is a child class of the Task Class,
+ * it specifies the task as a ToDo using [T]
+ */
+class ToDo extends Task {
+    public ToDo(String description) {
+        super(description);
+    }
+
+    @Override
+    public String toString() {
+        return "[T]" + super.toString();
+    }
+}
+
+/**
+ * The Deadline class is a child class of the Task Class,
+ * it specifies the task as a Deadline using [D]
+ */
+class Deadline extends Task {
+    protected Date deadlineBy;
+
+    public Deadline(String description, Date deadlineBy) {
+        super(description);
+        this.deadlineBy = deadlineBy;
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + "(by: " + deadlineBy + ")";
+    }
+}
+
+/**
+ * The Event class is a child class of the Task Class,
+ * it specifies the task as an Event using [E]
+ */
+class Event extends Task {
+    protected String at;
+
+    public Event(String description, String at) {
+        super(description);
+        this.at = at;
+    }
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + "(at:" + at + ")";
+    }
+}
+
+/**
+ * The DukeException is a child class of the Java Exception class
+ * which throws an exception message when an invalid input is given to the chat bot
+ * either a wrong message (unidentified input) or an unclear message (not specific enough)
+ */
+class DukeException extends Exception {
+    public DukeException(String message) {
+        super(message);
+    }
+
+    @Override
+    public String toString() {
+        return super.getMessage();
+    }
+}
```

Comment:
Perhaps you may want to separate the classes into different Java files? That would certainly declutter the amount of code in single Java file and makes it easier for developers to look at a specific area of code. Doing this would cause future extensions to increase the number of lines in a single class even more.